### DISCUSSION TITLE
Can Pulumi detect changes to a zipFile before deployment?

### DISCUSSION THREAD

**Discussion by amuresia** [pulumi/pulumi#7683] [Pulumi in Practice] [answered]
The `aws.synthetics.Canary` resource has an optional called `zipFile` that takes the code from a local path and deploys it. Can Pulumi detect changes to the code and deploy them as part of `pulumi up`?

**Comment by lukehoban** [comment, marked as answer]
We typically mark input properties like this as accepting a Pulumi `Archive`, which is a handle to file contents which can natively handle identifying changes and triggering updates only when changes are present.  We do this for `aws.lambda.Function` for example.  We likely should do the same here. Likely worth opening and issue (or a PR!) for this.  See https://github.com/pulumi/pulumi-aws/blob/master/provider/resources.go#L1788 for an example of how this is done for another resource.

**Reply by nimbinatus** [reply]
Opened as an issue: https://github.com/pulumi/pulumi/issues/9861
